### ISSUE TITLE
use trace ids from the client

### ISSUE THREAD

**Issue by Geal** [apollographql/router#297]
trace ids propagation works for spans generated in the router propagated to the subgraphs, but spans generated by the client are not used as parent of the router's root span.
Since there's some potential for abuse (clients using invalid spans or sampling values on open servers), this should be an option, deactivated by default.

Once we make a decision on this, supporting it is easy:

```diff
diff --git a/apollo-router/src/warp_http_server_factory.rs b/apollo-router/src/warp_http_server_factory.rs
index c797f97..7c18d0a 100644                                                                             
--- a/apollo-router/src/warp_http_server_factory.rs                                                       
+++ b/apollo-router/src/warp_http_server_factory.rs                                                       
@@ -286,14 +286,14 @@ where                                                                               
     PreparedQuery: graphql::PreparedQuery + 'static,                                                     
 {                                                                                                        
     // retrieve and reuse the potential trace id from the caller                                         
-    opentelemetry::global::get_text_map_propagator(|injector| {                                          
-        injector.extract_with_context(&Span::current().context(), &HeaderMapCarrier(&header_map));       
+    let parent_context = opentelemetry::global::get_text_map_propagator(|injector| {                     
+        injector.extract_with_context(&Span::current().context(), &HeaderMapCarrier(&header_map))        
     });                                                                                                  
                                                                                                          
     async move {                                                                                         
-        let response_stream = stream_request(router, request)                                            
-            .instrument(tracing::info_span!("graphql_request"))                                          
-            .await;                                                                                      
+        let span = tracing::info_span!("graphql_request");                                               
+        span.set_parent(parent_context);                                                                 
+        let response_stream = stream_request(router, request).instrument(span).await;                    
```

This should also add a test for this behavior, following up on #276 .

**Comment by abernix** [comment]
In theory this works, but needs to be tested.

**Comment by BrynCooke** [comment]
I think this mostly is fixed with #782 

Relevant information from [Otel Spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#sampling) 

In particular we now fulfill this requirements:
```
If there is a valid parent trace ID, use it. Otherwise generate a new trace ID (note: this must be done before calling ShouldSample, because it expects a valid trace ID as input).
```

TLDR the decision to sample is not made by the client unless [`ParentBased` sampling](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#parentbased) is enabled.
